src/GIFEncoder.js

```diff
@@ -472,6 +479,54 @@ GIFEncoder.prototype.ditherPixels = function (kernel, serpentine) {
   }
 
   var ds = kernels[kernel];
+  var direction = serpentine ? -1 : 1;
+  var index = y * width + x;
+
+  // Get original colour
+  var k = index * 3;
+  var r1 = frame[k];
+  var g1 = frame[k + 1];
+  var b1 = frame[k + 2];
+
+  // Get converted colour
+  // Gives me a value between 0 to 255 corresponding to index.
+  var idx = this.findClosestRGB(r1, g1, b1);
+  // Multiply by 3 because color tab will have r.g.b.
+  k = idx * 3;
+  var r2 = this.colorTab[k];
+  var g2 = this.colorTab[k + 1];
+  var b2 = this.colorTab[k + 2];
+
+  // calculate error.
+  var er = r1 - r2;
+  var eg = g1 - g2;
+  var eb = b1 - b2;
+
+  for (
+    var i = direction == 1 ? 0 : ds.length - 1,
+    end = direction == 1 ? ds.length : 0;
+    i !== end;
+    i += direction
+  ) {
+      var x1 = ds[i][1]; // *direction;  //  Should this by timesd by direction?..to make the kernel go in the opposite direction....got no idea....
+      var y1 = ds[i][2];
+      if (x1 + x >= 0 && x1 + x < width && y1 + y >= 0 && y1 + y < height) {
+        var d = ds[i][0];
+        k = index + x1 + y1 * width;
+        k *= 3;
+        // updated the error in data.
+        frame[k] = Math.max(0, Math.min(255, frame[k] + er * d));
+        frame[k + 1] = Math.max(0, Math.min(255, frame[k + 1] + eg * d));
+        frame[k + 2] = Math.max(0, Math.min(255, frame[k + 2] + eb * d));
+      }
+  }
+
+  return [idx, frame];
```

Comment:
Do we need to return frame and reassign? frame is passed by reference so it could be avoided.